### PR TITLE
CKS-278 Cache headers

### PR DESCRIPTION
**Pull request by 1996dylanriley**
https://nicedigital.atlassian.net/browse/CKS-278

### CODE REVIEW

File: web-app/CKS.Web.Test/CKS.Web.Test.csproj [hunk 1]
```diff
@@ -22,9 +22,36 @@
   </ItemGroup>
 
   <ItemGroup>
+    <None Update="appsettings.test.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (ediblecode) [inline web-app/CKS.Web.Test/CKS.Web.Test.csproj:26]: Should these all be the same `CopyToOutputDirectory`? Some are Always and some are PreserveNewest

File: web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs [hunk 2]
```diff
@@ -19,17 +19,30 @@ public class CKSWebApplicationFactory : WebApplicationFactory<Startup>
...
 		private readonly IList<Action<IServiceCollection>> serviceCollectionsActions = new List<Action<IServiceCollection>>();
+		private string webRootPath = null;
 
 		public CKSWebApplicationFactory(ITestOutputHelper output)
 		{
 			_output = output;
 			_output.WriteLine("CKSWebApplicationFactory.constructor");
 		}
 
+		protected override IWebHostBuilder CreateWebHostBuilder() => base.CreateWebHostBuilder();
```
Comment (ediblecode) [inline web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs:30]: We don't need this line if we're not calling `UseEnvironment` here. I'd rather we do:
```suggestion
		protected override IWebHostBuilder CreateWebHostBuilder() =>
    base.CreateWebHostBuilder().UseEnvironment("Testing");
```
(and remove line 36) as [that's what the docs say](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1#set-the-environment).

File: web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs [hunk 3]
```diff
@@ -19,17 +19,30 @@ public class CKSWebApplicationFactory : WebApplicationFactory<Startup>
...
+
 		protected override void ConfigureWebHost(IWebHostBuilder builder)
 		{
 			_output.WriteLine("CKSWebApplicationFactory.ConfigureWebHost");
 
+			builder.UseEnvironment("Testing");
```
Comment (ediblecode) [inline web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs:36]: The reverse comment from before - if we keep this here then we don't need line 30. Or we change line 30 to call `UseEnvironment` in which case we remove this line.

File: web-app/CKS.Web/Startup.cs [hunk 4]
```diff
@@ -77,7 +93,7 @@ public void ConfigureDevelopment(IApplicationBuilder app, IWebHostEnvironment en
 		{
 			// Serve static files straight for Gatsby's public folder when running locally.
 			// This means you don't have to copy the Gatsby output into the wwwroot folder like we do on TeamCity.
-			var localGatsbyFileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), "../../gatsby/public"));
+			var localGatsbyFileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), Configuration.GetValue<string>("GatsbyOutputFolder")));
```
Comment (ediblecode) [inline web-app/CKS.Web/Startup.cs:96]: We shouldn't need this app setting now that we've set the test environment. Please can you revert?

File: web-app/CKS.Web/appsettings.json [hunk 5]
```diff
@@ -7,5 +7,6 @@
     }
   },
   "ElasticSearchEnvironment": "CKSTest",
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "GatsbyOutputFolder": "../../gatsby/public"
```
Comment (ediblecode) [inline web-app/CKS.Web/appsettings.json:11]: Please can you remove this app setting?

File: web-app/CKS.Web/Startup.cs [hunk 6]
```diff
@@ -61,11 +58,30 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 					);
 
 			app.UseDefaultFiles();
-			app.UseStaticFiles();
-
 			app.UseRouting();
+            app.UseAuthorization();
+
+			app.UseStaticFiles(new StaticFileOptions
```
Comment (ediblecode) [inline web-app/CKS.Web/Startup.cs:64]: Can you put this back next to `UseDefaultFiles` please as they're related middleware - use defaultfiles means index.html is served by `UseStaticFiles`

File: web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs (hunk 3 above)
Comment (1996dylanriley) [reply web-app/CKS.Web.Test/IntegrationTests/Infrastructure/CKSWebApplicationFactory.cs:36]: interestingly if we use line 30 the environment is never actually set to testing.

File: web-app/CKS.Web/Startup.cs (hunk 4 above)
Comment (1996dylanriley) [reply web-app/CKS.Web/Startup.cs:96]: Yes, I realised this, however, its good practice to remove hardcoded into configuration files right?

Comment (ediblecode) [reply web-app/CKS.Web/Startup.cs:96]: generally but in this case it's *always* one level up as the gatsby folder is always in the same place and npm run build always outputs to the public folder so making it configurable just complicates things

**Pull request merged**